internal/controller/sharded.go

```diff
@@ -0,0 +1,57 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// StCtrlErrSet is the standard controller error container
+type StCtrlErrSet struct {
```

Comment:
could we use `type StCtrlErrSet map[string]error` ? And add functions to create this type.

Reply:
The reason behind using struct rather than map is that go doesn't support named parameters. Since Rec, Sync, and Update are all of the error type, if encapsulated into a function and the order of parameters is accidentally swapped, the compiler will not produce an error, but it may result in unexpected issues. By using a structure, users must explicitly specify field names, clearly expressing their intent.